openapi.go

```diff
@@ -536,3 +546,32 @@ func transformTypeName(s string) string {
 
 	return prefix + "_" + inside
 }
+
+type MyOperation struct {
+    *openapi3.Operation
+}
+
+func RegisterParameters(operation *openapi3.Operation,parameters ...*openapi3.Parameter) error {
+    if operation.Parameters == nil {
+        operation.Parameters = openapi3.Parameters{}
+    }
+
+    for _, param := range parameters {
+        if param == nil {
+            return errors.New("parameter cannot be nil")
+        }
+
+        for _, existing := range operation.Parameters {
+            if existing.Value != nil &&
+                existing.Value.Name == param.Name &&
+                existing.Value.In == param.In {
```

Comment:
You shouldn't check this. It's completely reasonable to have a cookie/header with a same name. A common case would be switching auth methods.